daemon/src/db.rs

```diff
@@ -545,103 +1593,152 @@ mod tests {
+            cfd_collaboratively_settled();
+        let order_id = cfd.id();
 
-        db.append_event(lock_confirmed(&cfd_final)).await.unwrap();
-        db.append_event(setup_failed(&cfd_final)).await.unwrap();
+        db.insert_cfd(&cfd).await.unwrap();
 
-        let cfd_ids = db.load_open_cfd_ids().await.unwrap();
-        assert!(cfd_ids.is_empty());
+        db.append_event(contract_setup_completed).await.unwrap();
+        db.append_event(collaborative_settlement_completed)
+            .await
+            .unwrap();
+        db.append_event(collab_settlement_confirmed(&cfd))
+            .await
+            .unwrap();
+
+        db.move_to_closed_cfds().await.unwrap();
+
+        let load_from_open = db.load_open_cfd::<DummyAggregate>(order_id, ()).await;
+        let load_from_closed = db.load_closed_cfd::<DummyAggregate>(order_id, ()).await;
+
+        assert!(load_from_open.is_err());
+        assert!(load_from_closed.is_ok());
     }
 
     #[tokio::test]
-    async fn given_order_rejected_then_do_not_load_non_final_cfd() {
+    async fn given_settlement_not_confirmed_when_move_cfds_to_closed_table_then_cannot_load_cfd_as_closed(
+    ) {
         let db = memory().await.unwrap();
 
-        let cfd_final = dummy_cfd();
-        db.insert_cfd(&cfd_final).await.unwrap();
+        let (cfd, contract_setup_completed, collaborative_settlement_completed) =
+            cfd_collaboratively_settled();
+        let order_id = cfd.id();
 
-        db.append_event(lock_confirmed(&cfd_final)).await.unwrap();
-        db.append_event(order_rejected(&cfd_final)).await.unwrap();
+        db.insert_cfd(&cfd).await.unwrap();
 
-        let cfd_ids = db.load_open_cfd_ids().await.unwrap();
-        assert!(cfd_ids.is_empty());
+        db.append_event(contract_setup_completed).await.unwrap();
+        db.append_event(collaborative_settlement_completed)
+            .await
+            .unwrap();
+
+        db.move_to_closed_cfds().await.unwrap();
+
+        let load_from_open = db.load_open_cfd::<DummyAggregate>(order_id, ()).await;
+        let load_from_closed = db.load_closed_cfd::<DummyAggregate>(order_id, ()).await;
+
+        assert!(load_from_open.is_ok());
+        assert!(load_from_closed.is_err());
     }
 
     #[tokio::test]
-    async fn given_final_and_non_final_cfd_then_non_final_one_still_loaded() {
+    async fn given_confirmed_settlement_when_move_cfds_to_closed_table_then_projection_aggregate_stays_the_same(
+    ) {
         let db = memory().await.unwrap();
 
-        let cfd_not_final = dummy_cfd();
-        db.insert_cfd(&cfd_not_final).await.unwrap();
+        let (cfd, contract_setup_completed, collaborative_settlement_completed) =
+            cfd_collaboratively_settled();
+        let order_id = cfd.id();
 
-        db.append_event(lock_confirmed(&cfd_not_final))
+        db.insert_cfd(&cfd).await.unwrap();
+
+        db.append_event(contract_setup_completed).await.unwrap();
```

Comment:
⚠️ I think these two `append_event` here don't make sense in the context of this test. The cfd in question is already settled, so why are we appending `contract_setup_completed` and `collaborative_settlement_completed`?
(It won't make a difference to the database, but it is confusing to read the test...)

Reply:
The helper `cfd_collaboratively_settled` returns a `Cfd` and a couple of events which when applied to the `Cfd` put it in a state where it is _then_ settled.

This is a requirement for the test to pass because:

- moving a CFD from `cfds` to `closed_cfds` fails if contract setup was not completed (we need a lock transaction).
- loading a closed CFD that doesn't have a `cet/collaborative_settlement/refund` transaction fails.

That's why we have to appends those two specific events to test this logic.